core/src/main/java/org/apache/iceberg/variants/VariantMetadata.java

```diff
@@ -34,4 +35,12 @@ public interface VariantMetadata extends Variants.Serialized {
 
   /** Returns the size of the metadata dictionary. */
   int dictionarySize();
+
+  /** Returns the serialized size in bytes of this value. */
+  int sizeInBytes();
+
+  /**
+   * Writes this value to the buffer at the given offset, ignoring the buffer's position and limit.
```

Comment:
Is it worth clarifying that this method should return how many bytes were written to the buffer?

Reply:
It is `sizeInBytes`. I can update if you think that is unclear.

Tests are passing now, so maybe we don't trigger a new CI run for this? I can update in a follow up.